examples/bearriver/src/FRP/BearRiver.hs

```diff
@@ -143,6 +143,24 @@ hold a = feedback a $ arr $ \(e,a') ->
 loopPre :: Monad m => c -> SF m (a, c) (b, c) -> SF m a b
 loopPre = feedback
 
+-- | Event source that never occurs.
+never :: Monad m => SF m a (Event b)
+never = MSF $ \_ -> return (NoEvent, never)
```

Comment:
```haskell
never = constant NoEvent
```

In general it's best not to use internal constructors if possible.

Using the API, as opposed to the internal constructors, facilitates changing the library internally without affecting uses/users. This is part of the reason for an API in the first place.

While this often creates some (small) performance loss, it's worth it when libraries grow big, since you can change the internals quickly. For example, if you choose to use GADTs to improve performance, and implement `constant` using that optimisation, you'll only get a gain if you stuck to the API. For another example, if you choose to add another field or change the order of the elements in the tuple, you won't have to change everything.

Reply:
Yes, I understand.
Saw that you use the internal MSF constructor a couple of times and assumed it is OK in such cases. But of course when a function can be implemented using the existing API then it should always go that way - but sometimes it is not clear for me to see that but thanks for the help.